wasmtime-jit/src/compiler.rs

```diff
@@ -309,6 +321,24 @@ fn allocate_functions(
     Ok(result)
 }
 
+fn register_traps(
+    allocated_functions: &PrimaryMap<DefinedFuncIndex, *mut [VMFunctionBody]>,
+    traps: &Traps,
+    trap_registration_guards: &mut Vec<TrapRegistrationGuard>,
+) {
+    let mut trap_registry = get_mut_trap_registry();
+    for (func_addr, func_traps) in allocated_functions.values().zip(traps.values()) {
+        for trap_desc in func_traps.iter() {
+            let func_addr = *func_addr as *const u8;
+            let offset = usize::try_from(trap_desc.code_offset).unwrap();
+            let trap_addr = unsafe { func_addr.add(offset) };
```

Comment:
Could we eliminate this `unsafe` by casting `func_addr` to `usize` here, so we can just do a plain add, rather than doing it inside of `register_trap`?

Reply:
It's a good idea.